snapshot-labs/mana
# feat(starknet): allow custom per-space accounts from mnemonic

Closes: https://github.com/snapshot-labs/mana/issues/123

This PR replaces address + private key combo with mnemonic that can be used to derive per-space addresses.

At the same time I created two new accounts on both mainnet and goerli under indexes 1 and 2.
1 is default, 2 could be used for Starknet Governance. 0 is omitted because it was legacy Cairo 0 address that was deployed via proxy and derivation is different from new accounts so I just omitted it so we don't have to handle different derivations for each.

## Test plan
1. Run mana from this branch as is, it should work.
2. Check that transactions go from `0x04d1815453874b0B0DdEbFf592485f151b9886Dee32AFd404846c318F54fB161`.
3. Change address in addressIndicies to one of your spaces.
4. Actions on this space go from `0x00462fd68DD760983e233dE784eFC402F91a36AE9B24c7A5a520a5CB3F0fE4d6`.
